File: desafio-04/rafaeldss/xadrez.py

tabuleiro = [
    [0, 0, 0, 0, 2, 0, 2, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 4, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 1, 1, 0, 0, 0],
    [0, 0, 0, 1, 1, 0, 0, 0],
    [0, 5, 0, 4, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0]
    ]

qt_pecas = {1:[0, 'Peão'],
            2:[0, 'Bispo'],
            3:[0, 'Cavalo'],
            4:[0, 'Torre'],
            5:[0, 'Rainha'],
            6:[0, 'Rei']
            }

for row in tabuleiro:
    for peca in row:
        if peca != 0:
            qt_pecas[peca][0] += 1
                
for val in qt_pecas.values():
    print(f'{val[1]}: {val[0]} peça(s)')
    

File: desafio-04/rafaeldss/c/xadrez.c
#include <stdio.h>


int main(void) {
    char nome_pecas[][7] = {"", "Peão", "Bispo", "Cavalo", "Torre", "Rainha", "Rei"};
    int qt_pecas[7] = {0};
    int tabuleiro[][8] = {
        {4, 3, 2, 5, 6, 2, 3, 4},
        {1, 1, 1, 1, 1, 1, 1, 1},
        {0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0},
        {0, 0, 0, 0, 0, 0, 0, 0},
        {1, 1, 1, 1, 1, 1, 1, 1},
        {4, 3, 2, 5, 6, 2, 3, 4}
    };

    for (int index_linha = 0; index_linha < 8; index_linha++) {
        for (int index_peca = 0; index_peca < 8; index_peca++) {
            int peca_valor = tabuleiro[index_linha][index_peca];
            qt_pecas[peca_valor]++;
        }
    }

    for (int i = 1; i < 7; i++) {
        printf("%s: %d peça(s)\n", nome_pecas[i], qt_pecas[i]);
    }
}
